### ISSUE TITLE
[Bug] Crash when open a non-english Folder

### ISSUE THREAD

**Issue by KAIYOHUGO** [pop-os/cosmic-files#241]
When I open a folder that its name is not English (or specific, Chinese), it crash. 
It can show Chinese text correctly, and open normal file is also fine.

# Output

```bash
❯ RUST_BACKTRACE=1 cosmic-files

[2024-06-12T07:05:52Z WARN  wgpu_hal::gles::adapter] Detected skylake derivative running on mesa i915. Clears to srgb textures will use manual shader clears.
[2024-06-12T07:05:52Z WARN  wgpu_hal::gles::adapter] Detected skylake derivative running on mesa i915. Clears to srgb textures will use manual shader clears.
[2024-06-12T07:05:52Z WARN  cosmic_text::font::system] failed to load font 'SymbolsNFM'
thread 'main' panicked at /build/cosmic-files-5r9a5g/cosmic-files-0.1.0~1717593356~22.04~c639d6d/vendor/sctk-adwaita/src/title/ab_glyph_renderer.rs:104:40:
index out of bounds: the len is 6080 but the index is 6278
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread 'main' panicked at /build/cosmic-files-5r9a5g/cosmic-files-0.1.0~1717593356~22.04~c639d6d/vendor/winit/src/platform_impl/linux/wayland/window/mod.rs:471:14:
called `Result::unwrap()` on an `Err` value: PoisonError { .. }
stack backtrace:
   0:     0x5c1f6508a2ab - <unknown>
   1:     0x5c1f650cfbd0 - <unknown>
   2:     0x5c1f6507d3ed - <unknown>
   3:     0x5c1f6508a094 - <unknown>
   4:     0x5c1f650b2f0a - <unknown>
   5:     0x5c1f650b2bad - <unknown>
   6:     0x5c1f650b3328 - <unknown>
   7:     0x5c1f6508a68e - <unknown>
   8:     0x5c1f6508a4c6 - <unknown>
   9:     0x5c1f650b3052 - <unknown>
  10:     0x5c1f64139fd5 - <unknown>
  11:     0x5c1f6413a783 - <unknown>
  12:     0x5c1f647493cf - <unknown>
  13:     0x5c1f642f9c54 - <unknown>
  14:     0x5c1f642a3f06 - <unknown>
  15:     0x5c1f642a3a70 - <unknown>
  16:     0x5c1f642bc821 - <unknown>
  17:     0x5c1f6424a767 - <unknown>
  18:     0x5c1f6425daf6 - <unknown>
  19:     0x5c1f641aee9f - <unknown>
  20:     0x5c1f641b09ab - <unknown>
  21:     0x5c1f6426f240 - <unknown>
  22:     0x5c1f6417f8c9 - <unknown>
  23:     0x5c1f6413abc3 - <unknown>
  24:     0x5c1f6413ac01 - <unknown>
  25:     0x5c1f650b2f44 - <unknown>
  26:     0x5c1f65098e6b - <unknown>
  27:     0x5c1f6413abee - <unknown>
  28:     0x7d344d029d90 - __libc_start_call_main
                               at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  29:     0x7d344d029e40 - __libc_start_main_impl
                               at ./csu/../csu/libc-start.c:392:3
  30:     0x5c1f6413aaf5 - <unknown>
  31:                0x0 - <unknown>
thread 'main' panicked at library/core/src/panicking.rs:144:5:
panic in a destructor during cleanup
thread caused non-unwinding panic. aborting.
```

# Environment

```
OS: Pop!_OS 22.04 LTS x86_64
Host: Delbin rev4
Kernel: 6.8.0-76060800daily20240311-generic
Resolution: 1920x1080
DE: GNOME 42.9
WM: Mutter
WM Theme: Pop
Theme: Pop-dark [GTK2/3]
Icons: Pop [GTK2/3]
Terminal: alacritty
CPU: 11th Gen Intel i5-1135G7 (8) @ 4.200GHz
GPU: Intel TigerLake-LP GT2 [Iris Xe Graphics]
```

**Comment by Friz64** [comment]
Opened https://github.com/PolyMeilex/sctk-adwaita/issues/63

**Comment by PolyMeilex** [comment]
Published sctk-adwaita 0.8.3 that should now ignore the extra pixels rather than panic, so cargo update should do the trick. 

On a different note, Cosmic apps should probably just disable this feature of winit, as you have your own decorations anyway I suppose.

**Comment by KAIYOHUGO** [comment]
I can confirm that this has been fixed in latest build. Thanks!
